test/functional/feature_discover.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -discover command."""
+
+import socket
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_aton(address)
+    except socket.error:
+        return False
+    return True
+
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:
+        return False
+    return True
+
+
+class DiscoverTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def validate_addresses(self, addresses_obj):
+        for address_obj in addresses_obj:
+            address = address_obj['address']
+            self.log.info(f"Validating {address}")
+            valid = (is_valid_ipv4_address(address)
+                     or is_valid_ipv6_address(address))
+            assert_equal(valid, True)
+
+    def test_local_addresses(self, test_case, *, expect_empty=False):
+        self.log.info(f"Restart node with {test_case}")
+        self.restart_node(0, test_case)
+        network_info = self.nodes[0].getnetworkinfo()
+        network_enabled = [n for n in network_info['networks']
+                           if n['reachable'] and n['name'] in ['ipv4', 'ipv6']]
+        local_addrs = list(network_info["localaddresses"])
+        if expect_empty or not network_enabled:
+            assert_equal(local_addrs, [])
+        elif len(local_addrs) > 0:
+            self.validate_addresses(local_addrs)
+
+    def run_test(self):
+        test_cases = [
+            ["-listen", "-discover"],
+            ["-listen=1", "-discover"],
+            ["-listen", "-discover=1"],
+            ["-listen=1", "-discover=1"],
+            ["-discover"],
+        ]
```

Comment:
Are there any other occasions where we test the same behavior for different variations of an argument? Testing with combinations of `-discover`, `-discover=1` or `-listen`, `-listen=1` seems redundant. Taking that into consideration, I  think that this test could be replaced with just a test case for `discover=0` at `test/functional/feature_bind_port_discover.py`.

Reply:
It was suggested by other reviewer, but I agree with you, there is no need to test redundant cases.

Reply:
Done!